llvm/lib/Target/BPF/BPFAsmPrinter.cpp

```diff
@@ -59,7 +60,9 @@ bool BPFAsmPrinter::doInitialization(Module &M) {
   AsmPrinter::doInitialization(M);
 
   // Only emit BTF when debuginfo available.
-  if (MAI->doesSupportDebugInformation() && !M.debug_compile_units().empty()) {
+  // Unsupported for Solana: https://github.com/solana-labs/llvm-project/issues/37
+  if (MAI->doesSupportDebugInformation() && !M.debug_compile_units().empty() &&
+      TM.getMCSubtargetInfo()->hasFeature(BPF::FeatureSolana) || TM.getTargetTriple().getArch() == Triple::sbf) {
```

Comment:
Shouldn't the condition be negated? `!TM.getMCSubtargetInfo()->hasFeature(BPF::FeatureSolana) && TM.getTargetTriple().getArch() != Triple::sbf`?  I thought we don't want to emit BTF for Solana.

Reply:
Oh, of course. Sorry